### DISCUSSION TITLE
why no custom logs output in my terminal

### DISCUSSION THREAD

**Discussion by jaffe-fly** [fastapi/fastapi#8517] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# my main.py
import uvicorn
from fastapi import FastAPI

import logging
from anomaly import anomal

app = FastAPI(description="运用大数据及AI算法，对资源池进行容量预测，对虚机进行标签分类，而后进行重组装箱。",
              title="云效管家系统",
              version="v1.0.0",
              )


app.include_router(anomal, prefix="/api")


@app.get("/")
def read_root():
    logging.info("test ok")
    return {"Hello": "World"}


if __name__ == '__main__':
    uvicorn.run('main:app', host='0.0.0.0', port=8000, reload=True, debug=True, workers=1)


# my test.py

from fastapi.testclient import TestClient

from main import app

client = TestClient(app)
def test_read_main():
    response = client.get("http://127.0.0.1:8000/")
    assert response.status_code == 200
    assert response.json() == {"Hello": "World"}

def test_anomaly():
    postdata = {'jobId': 1,
                'jobType': 'anomaly',
                'targetClass': 'vm',
                'targetId': 123,
                'params': {'perf_data': [...]},
                }
    response = client.post('http://127.0.0.1:8000/api/getAnomaly', json=postdata,
                           headers={'Content-Type': 'application/json'})
    print(response.json())
    assert response.status_code == 200
```

### Description

when i run my `main.py` file ,the logs is 
![image](https://user-images.githubusercontent.com/49515380/172604309-00869375-dff8-4d8d-a696-2a7b79508d3c.png)

when i run `test_read_main() `in the `test.py `,
no other logs output,no `"test ok"` log output

when i run `test_anomaly()` func, `anomal` is a `APIRouter()`,also no logs output
in other words, No matter how I test ，the logs  in terminal No change at all
why???

i searched this on internet ,but no method solved this

could anyone help me ？thank you very much

### Operating System

Linux, Windows

### FastAPI Version

0.78.0

### Python Version

3.9.7

**Comment by jaffe-fly** [comment]
my `anomaly` is

```
from fastapi import APIRouter, HTTPException
from logzero import logger
from model import AnomalReturn, AnomalReceive

anomal = APIRouter()


@anomal.post("/getAnomaly", response_model=AnomalReturn)
def anomaly(data: AnomalReceive):
    perf_data = [i.dict() for i in data.params.perf_data]
    jobType = data.jobType
    targetId = data.targetId
    logger.info(f"jobId={data.jobId},targetId={targetId}, datalen={len(perf_data)}")
    return {"jobId": data.jobId, "targetId": targetId, "message": message, "result": result}
```
this logger also no output

**Comment by JarroVGIT** [comment]
You are calling your FastAPI app with the `TestClient`, and not via HTTP. If you were to run your server and access the root path, you would see the logging as expected. I don't know if the TestClient can take a `debug=True` param.

**Comment by odiseo0** [comment]
Also, I think you use a custom logger directly with [Uvicorn](url) if I remember correctly

**Comment by Kludex** [comment]
👋 🏃 💨 
```python
import logging

from fastapi import FastAPI

app = FastAPI()


logger = logging.getLogger(__name__)
logger.setLevel(logging.INFO)
logger.addHandler(logging.StreamHandler())


@app.get("/")
def home():
    logger.info("Hello World")
```

**Comment by jaffe-fly** [comment]
> You are calling your FastAPI app with the `TestClient`, and not via HTTP. If you were to run your server and access the root path, you would see the logging as expected. I don't know if the TestClient can take a `debug=True` param.

when i use requests to test like this
```
import requests

r = requests.get("http://127.0.0.1:8000/")
print(r.json())
```
also no log out "test ok"

**Comment by jaffe-fly** [comment]
> 👋 🏃 💨
> 
> ```python
> import logging
> 
> from fastapi import FastAPI
> 
> app = FastAPI()
> 
> 
> logger = logging.getLogger(__name__)
> logger.setLevel(logging.INFO)
> logger.addHandler(logging.StreamHandler())
> 
> 
> @app.get("/")
> def home():
>     logger.info("Hello World")
> ```

 **pytest or request to test this also no logs out,its alwayse**

![image](https://user-images.githubusercontent.com/49515380/172741867-f00aab89-415f-4440-b7c1-a606d3f514d7.png)

**Comment by Kludex** [comment]
With the latest uvicorn (0.17.6) it works just fine here.

**Comment by yusuken1983** [comment]
Please use `"uvicorn"` instead of `__name__` .

```python
logger = logging.getLogger("uvicorn")

@app.get("/")
def home():
    logger.info("Hello World")
```

**Reply by nicataliyev** [reply]
This helped thanks @yusuken1983

**Comment by jamal-alkharrat** [comment]
I had the same error. It was happening because i had 2 Terminals with uvicorn running. when i closed the second one, it then worked fine for me.
